src/ejabberd_sm.erl

```diff
@@ -247,6 +248,26 @@ store_info(User, Server, Resource, {Key, _Value} = KV) ->
             end
     end.
 
+-spec remove_info(jid:user(), jid:server(), jid:resource(),
+                 {any(), any()}) -> ok | {error, offline}.
+remove_info(User, Server, Resource, Key) ->
+    case get_session(User, Server, Resource) of
+        offline -> {error, offline};
+        {_SUser, SID, SPriority, SInfo} ->
+            case SID of
+                {_, Pid} when self() =:= Pid ->
+                    %% It's safe to allow process update it's own record
```

Comment:
Minor: 'allow the process to update its'

Reply:
Done